src/kindtech/geo/_arcgis/_ingestion.py

```diff
@@ -0,0 +1,185 @@
+"""
+ArcGIS REST Services ingestion module.
+
+This module provides functions to load ArcGIS REST Services data from the web and save
+it to a CSV file
+
+The CSV file is later used as the main lookup table to find the relevant data
+"""
+
+import calendar
+import logging
+from pathlib import Path
+from typing import Any
+
+import polars as pl
+import requests
+from bs4 import BeautifulSoup
+
+# Set up logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+# Constants
+ARCGIS_BASE_URL = "https://services1.arcgis.com/ESMARspQHYMw9BZ9/arcgis/rest/services/"
+DEFAULT_OUTPUT_PATH = Path(__file__).parent / "data" / "arcgis_services.csv"
+
+# Lookup tables for parsing
+MONTH = {}
+for idx, name in enumerate(calendar.month_name):  # '' then 'January'…
+    if name:
+        MONTH[name.upper()[:3]] = idx  # JAN, FEB, …
+        MONTH[name.upper()] = idx  # JANUARY, FEBRUARY …
+
+REGION = {"UK", "GB", "EW"}
+RES = {"BFC", "BFE", "BGC", "BSC"}
+DROP = {
+    "LOCAL",
+    "AUTHORITY",
+    "DISTRICT",
+    "DISTRICTS",
+    "BOUNDARY",
+    "BOUNDARIES",
+    "WITH",
+    "AND",
+    "IN",
+}
+
+
+def _load_arcgis_data(url: str = ARCGIS_BASE_URL) -> str:
```

Comment:
Here should we use the logic similar to huggingface where we don't redownload the raw data if they already exists locally?

Reply:
Good idea, same as with the logging I see this as an improvement that we can implement later, sounds good?